imap/search_query.c

```diff
@@ -724,7 +748,9 @@ static void subquery_run_indexed(const char *key __attribute__((unused)),
 
     bx = search_begin_search(query->state->mailbox, opts);
     if (!bx) {
-        r = IMAP_INTERNAL;
+        r = config_getenum(IMAPOPT_SEARCH_ENGINE) == IMAP_ENUM_SEARCH_ENGINE_SQUAT ?
+            subquery_run_one_folder(query, index_mboxname(query->state), sub->indexed) :
+            IMAP_INTERNAL;
```

Comment:
The ternary is hard to read here -- there's a lot of text to read for a single statement.  What about something like:

```suggestion
        if (config_getenum(IMAPOPT_SEARCH_ENGINE) == IMAP_ENUM_SEARCH_ENGINE_SQUAT) {
            r = subquery_run_one_folder(query, index_mboxname(query->state), sub->indexed);
        }
        else {
            r = IMAP_INTERNAL;
        }
```

or maybe even just:

```suggestion
        r = IMAP_INTERNAL;
        if (config_getenum(IMAPOPT_SEARCH_ENGINE) == IMAP_ENUM_SEARCH_ENGINE_SQUAT)
            r = subquery_run_one_folder(query, index_mboxname(query->state), sub->indexed);
```